File: frontend/src/api/mastery.ts
import { apiFetch } from '@/api/client'
import type { MasteryProfile } from '@/types/api'

export const getMastery = (windowDays?: number) =>
  apiFetch<MasteryProfile>(
    windowDays != null ? `/users/me/mastery?window_days=${windowDays}` : '/users/me/mastery',
  )

File: backend/api/mastery.py
from __future__ import annotations

from fastapi import APIRouter, Depends

from backend.deps import current_user
from backend.services import ai_gateway
from shared.schemas import MasteryProfile, User

router = APIRouter(prefix="/users/me", tags=["mastery"])


@router.get("/mastery", response_model=MasteryProfile)
async def mastery(window_days: int | None = None, user: User = Depends(current_user)) -> MasteryProfile:
    return ai_gateway.build_profile(user.id, window_days)
